src/ShellyDuo/shelly_init.cpp

```diff
@@ -49,6 +49,14 @@ void CreateComponents(std::vector<std::unique_ptr<Component>> *comps,
     return;
   }
 
+  // Use adaptive lightning when possible (CCT)
+  std::unique_ptr<hap::AdaptiveLighting> adaptive_light;
+  adaptive_light.reset(new hap::AdaptiveLighting(hap_light.get(), lb_cfg));
+  auto st = adaptive_light->Init();
+  if (st.ok()) {
+    hap_light->SetAdaptiveLight(std::move(adaptive_light));
+  }
+
```

Comment:
Could this be integrated in the `CCTController`? Because it's the only light which supports it.

Reply:
I guess it could. But on the other hand: the controller should not need to know about adaptive lights existing. And the Adaptive Lightning Protocol of Apple is generic in the way that it could support autotuning other characteristics, not limited to only color temperature, so I actually think it would make more sense this way

Reply:
Good argument. My main point is that I don't like ot have redundant code in the device specific files. You follow the current pattern, so it's ok. That problem should be solved, but not in this PR.

Reply:
agree